### PR TITLE
Implement updates to the API Programmer's Guide online content + PDF

### LINKED ISSUE #16289: Implement updates to the API Programmer's Guide online content + PDF
## Problem statement
To address user feedback, we updated some content on the API Programmer's Guide. See #16106 and #15997 for reference.

## What you need to know
The website needs to be updated using the content doc linked below.
- [Content doc](https://cdc.sharepoint.com/:w:/r/teams/ReportStream/Shared%20Documents/Design%20and%20Research/03_Content/Website%20content/Programmer%27s%20Guide%20updates/20241018_API_guide_content_updates.docx?d=w4563ca769bb840bdb617890ba194c4c0&csf=1&web=1&e=dCvttH)

## Acceptance criteria

- [x] Web content updated 
- [x] PDF download updated 
- [x] Tag @jillian-hammer in PR

### PR DESCRIPTION
**Pull request by jpandersen87**
Fixes #16289

This PR makes the requested content changes, in addition to:

- Moves all usages of smartsheet, one pager pdf, and nist validator tool urls to site.json
- Removes references to "Sample payloads and output" page from app router and sitemap (code kept as it may return)
  - E2E test is skipped

### CODE REVIEW

**Comment by jpandersen87** [comment]
@jillian-hammer should be ready for review on trialfrontend01 once it deploys

File: frontend-react/src/content/developer-resources/reportstream-api/getting-started/GettingStarted.mdx
Comment (jillian-hammer) [inline frontend-react/src/content/developer-resources/reportstream-api/getting-started/GettingStarted.mdx:45]: "NIST HL7 General Validation Tool" should be hyperlinked to: `https://hl7v2-gvt.nist.gov/gvt/#/home`

**Review by jillian-hammer — CHANGES_REQUESTED** [review]
thanks @jpandersen87 - these changes look great. I caught one missing link. once that is changed, it's good to merge from a design pov!

**Review by jillian-hammer — APPROVED** [review]
thanks for adding the missing link - LGTM @jpandersen87

**Review by victor-chaparro — APPROVED** [review]
LGTM

**Pull request merged**
